### PR TITLE
utils.get_class_field_meta

### LINKED ISSUE #34: Implement ItemClassAdapter
This would help implementing delayed item object creation in `itemloaders`: https://github.com/scrapy/itemloaders/pull/20#issuecomment-681867612

It should only accept item classes, not objects.

It should implement `get_field_meta` just as `ItemAdapter`.

### PR DESCRIPTION
**Pull request by elacuesta**
Closes #34

Current tests pass with this change, ~I'll add some dedicated tests shortly~. Naming suggestions are welcome.

### CODE REVIEW

File: itemadapter/utils.py [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+
+def _is_dataclass_subclass(obj: Any) -> bool:
+    try:
+        import dataclasses
+    except ImportError:
+        return False
+    return dataclasses.is_dataclass(obj)
+
+
+def _is_attrs_subclass(obj: Any) -> bool:
```
Comment (kmike) [inline itemadapter/utils.py:34]: Users are not subclassing from attrs or dataclasses - what do you think about naming it `_is_attrs_class`, `_is_dataclass`?

File: itemadapter/utils.py [hunk 2]
```diff
@@ -56,3 +86,37 @@ def is_item(obj: Any) -> bool:
         or is_dataclass_instance(obj)
         or is_attrs_instance(obj)
     )
+
+
+def get_class_field_meta(item_class: type, field_name: str) -> MappingProxyType:
```
Comment (kmike) [inline itemadapter/utils.py:91]: The name looks good. We can also name it get_field_meta, but no strong preference here.

Comment (Gallaecio) [reply itemadapter/utils.py:91]: I would keep the `class` there, leave `get_field_meta` for a possible future function that accepts both items and item classes.

File: itemadapter/utils.py [hunk 3]
```diff
@@ -1,6 +1,45 @@
...
 from typing import Any
 
 
+__all__ = [
+    "get_class_field_meta",
+    "is_attrs_instance",
+    "is_dataclass_instance",
+    "is_item",
+    "is_scrapy_item",
+]
```
Comment (Gallaecio) [inline itemadapter/utils.py:11]: +0

If someone uses an asterisk import and uses _-prefixed functions imported that way, they deserve whatever happens to them :slightly_smiling_face:

**Review by Gallaecio — APPROVED** [review]

File: itemadapter/utils.py (hunk 2 above)
Comment (elacuesta) [reply itemadapter/utils.py:92]: On second thought, is `get_field_meta_from_class` too verbose?

Comment (Gallaecio) [reply itemadapter/utils.py:92]: It’s not too verbose to me. And it’s slightly less ambiguous.

**Comment by elacuesta** [comment]
@Gallaecio please check again after a6af34a, I updated the docs and made some changes so the function fails for non item classes.

File: itemadapter/utils.py (hunk 3 above)
Comment (elacuesta) [reply itemadapter/utils.py:11]: hahaha fair enough

**Review by kmike — APPROVED** [review]
👍

File: itemadapter/__init__.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 from .adapter import ItemAdapter  # noqa: F401
-from .utils import is_item  # noqa: F401
+from .utils import is_item, get_class_field_meta  # noqa: F401
```
Comment (Gallaecio) [inline itemadapter/__init__.py]: :lipstick: 

```suggestion
from .utils import get_class_field_meta, is_item  # noqa: F401
```

**Review by Gallaecio — APPROVED** [review]
Looks good to me, feel free to merge.

**Pull request merged**